### PR TITLE
ELE-423: Emit Search requests when clicking Autocomplete terms

### PR DESCRIPTION
**Pull request by geoffcoutts**
Clicking on an autocomplete term will now emit an event (`gbe::update_search_term`) for updating the searchbox value and emitting a search request. Additionally, this event can be reused elsewhere and includes a `search` property that can be set to false if the desire is to not emit a search.

### CODE REVIEW

File: packages/web-components/@groupby/elements-autocomplete/test/unit/autocomplete.test.ts [hunk 1]
```diff
@@ -432,4 +433,26 @@ describe('Autcomplete Component', () => {
...
+        group,
+        search: true,
+      };
+      const dispatchElementsEvent = stub(autocomplete, 'dispatchElementsEvent');
+
+      autocomplete.sendUpdateSearchEvent(clickEvent);
+
+      expect(dispatchElementsEvent).to.be.calledWith(UPDATE_SEARCH_TERM, payload);
+    });
+  });
```
Comment (ChantalDesrochers) [inline packages/web-components/@groupby/elements-autocomplete/test/unit/autocomplete.test.ts]: Is this dispatching an UPDATE_SEARCH_TERM event rather than sending a search request as the test description states?

File: packages/web-components/@groupby/elements-autocomplete/test/unit/autocomplete.test.ts [hunk 2]
```diff
@@ -432,4 +433,26 @@ describe('Autcomplete Component', () => {
       expect(autocomplete.selectedIndex).to.equal(newSelectedIndex);
     });
   });
+
+  describe('sendUpdateSearchEvent()', () => {
+    it('should send a search request with a search term', () => {
```
[the remark this answers is not in the document]
Comment (ChantalDesrochers) [reply packages/web-components/@groupby/elements-autocomplete/test/unit/autocomplete.test.ts]: or, actually maybe not, since the function always sets the search boolean to true.

File: packages/web-components/@groupby/elements-search-box/test/unit/search-box.test.ts [hunk 3]
```diff
@@ -146,37 +146,59 @@ describe('SearchBox Component', () => {
...
       searchbox.group = group;
 
-      searchbox.updateText(inputEvent);
+      searchbox.updateSearch(inputEvent);
 
       expect(updateSearchTermValue).to.be.calledWith(term);
     });
 
-    it('should not update the value property with data from the event when the group in the component and the event do not match', () => {
+    it('should emit a search request if event requests it and group matches', () => {
```
Comment (ChantalDesrochers) [inline packages/web-components/@groupby/elements-search-box/test/unit/search-box.test.ts]: maybe reword to something like '...if event payload has search property set to true...'

File: packages/web-components/@groupby/elements-search-box/test/unit/search-box.test.ts [hunk 4]
```diff
@@ -146,37 +146,59 @@ describe('SearchBox Component', () => {
...
       searchbox.group = 'different group';
 
-      searchbox.updateText(inputEvent);
+      searchbox.updateSearch(inputEvent);
 
       expect(updateSearchTermValue).to.not.be.called;
+      expect(emitSearchEvent).to.not.be.called;
+    });
+
+    it('should not emit a search request if group matches but event does not request it', () => {
```
Comment (ChantalDesrochers) [inline packages/web-components/@groupby/elements-search-box/test/unit/search-box.test.ts]: same suggestion as above re. wording.

File: packages/web-components/@groupby/elements-autocomplete/test/unit/autocomplete.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (geoffcoutts) [reply packages/web-components/@groupby/elements-autocomplete/test/unit/autocomplete.test.ts]: Yeha, that's why the flag wasn't specified.

File: packages/web-components/@groupby/elements-search-box/test/unit/search-box.test.ts (hunk 3 above)
Comment (geoffcoutts) [reply packages/web-components/@groupby/elements-search-box/test/unit/search-box.test.ts]: we didn't make it something much longer like that because we were trying to reduce line lengths for test descriptions. This also has the benefit of not being too formal while having the details in the test.

File: packages/web-components/@groupby/elements-autocomplete/README.md [hunk 5]
```diff
@@ -44,6 +44,11 @@ This component dispatches a number of events. These events are defined in the [`
 
 This event is dispatched when one of the search terms inside of the `gbe-autocomplete` component is put in the active state or hovered on.
 
+#### `UPDATE_SEARCH_TERM`
+
+This event is dispatched when one of the search terms inside of the `gbe-autocomplete` component is clicked.
```
[the remark this answers is not in the document]
Comment (geoffcoutts) [reply packages/web-components/@groupby/elements-autocomplete/README.md]: I have updated references of `search term` to `autocomplete term` in this package.

File: packages/web-components/@groupby/elements-autocomplete/test/unit/autocomplete.test.ts (hunk 1 above)
Comment (geoffcoutts) [reply packages/web-components/@groupby/elements-autocomplete/test/unit/autocomplete.test.ts]: Updated the test description to better reflect that.
